File: customLoadingDialog/src/main/java/com/saksham/customloadingdialog/Loader.java
package com.saksham.customloadingdialog;

import android.content.Context;

public class Loader {
    static CustomLoader customLoader;

    public static void showDialog(Context context, boolean cancellable, int animation){
        hideDialog();
        if(context != null){
            try {
                customLoader = new CustomLoader(context, animation);
                customLoader.setCanceledOnTouchOutside(true);
                customLoader.setCancelable(cancellable);
                customLoader.show();
            } catch (Exception e){
                e.printStackTrace();
            }
        }
    }

    public static void hideDialog(){
        if(customLoader!=null && customLoader.isShowing()){
            try{
                customLoader.dismiss();
                customLoader = null;
            }catch (Exception e){
                customLoader = null;
            }
        }
    }
}

File: customLoadingDialog/src/main/java/com/saksham/customloadingdialog/Loader.kt
package com.saksham.customloadingdialog

import android.content.Context

private var customLoader: CustomLoader? = null
fun showDialog(context: Context?, cancelable: Boolean, animation: Int) {
    hideDialog()
    if (context != null) {
        try {
            customLoader = CustomLoader(context, animation)
            customLoader?.let {
                it.setCanceledOnTouchOutside(true)
                it.setCancelable(cancelable)
                it.show()
            }

        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}

fun hideDialog() {
    if (customLoader != null && customLoader?.isShowing!!) {
        customLoader = try {
            customLoader?.dismiss()
            null
        } catch (e: Exception) {
            null
        }
    }
}